parquet/src/arrow/async_reader/mod.rs

```diff
@@ -628,8 +629,6 @@ where
                     )
                     .await?;
 
-                let mut cache_projection = predicate.projection().clone();
```

Comment:
The overall set of projections to cache is already calculated at the start of the loop -- caching the subset of columns individually again for each predicate seems unecessary